From be29c837230519162c5731b175213463fbf1f10f Mon Sep 17 00:00:00 2001
From: Patrick Steinhardt <ps@pks.im>
Date: Mon, 26 Oct 2015 16:21:09 +0100
Subject: [PATCH 1011/1024] repository: expose `repo_init_create_head`

Expose the function `repo_init_create_head` as
`git_repository_create_head`.
---
 src/repository.c | 4 ++--
 src/repository.h | 1 +
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/repository.c b/src/repository.c
index 7e1e05d..34154b8 100644
--- a/src/repository.c
+++ b/src/repository.c
@@ -1040,7 +1040,7 @@ static int check_repositoryformatversion(git_config *config)
 	return 0;
 }
 
-static int repo_init_create_head(const char *git_dir, const char *ref_name)
+int git_repository_create_head(const char *git_dir, const char *ref_name)
 {
 	git_buf ref_path = GIT_BUF_INIT;
 	git_filebuf ref = GIT_FILEBUF_INIT;
@@ -1737,7 +1737,7 @@ int git_repository_init_ext(
 				repo_path.ptr, wd, opts)) &&
 			!(error = repo_init_config(
 				repo_path.ptr, wd, opts->flags, opts->mode)))
-			error = repo_init_create_head(
+			error = git_repository_create_head(
 				repo_path.ptr, opts->initial_head);
 	}
 	if (error < 0)
diff --git a/src/repository.h b/src/repository.h
index c8e89af..bbdd883 100644
--- a/src/repository.h
+++ b/src/repository.h
@@ -153,6 +153,7 @@ GIT_INLINE(git_attr_cache *) git_repository_attr_cache(git_repository *repo)
 }
 
 int git_repository_head_tree(git_tree **tree, git_repository *repo);
+int git_repository_create_head(const char *git_dir, const char *ref_name);
 
 /*
  * Weak pointers to repository internals.
-- 
2.8.1.windows.1

